packages/utils/types/NodePath.ts

```diff
@@ -0,0 +1,76 @@
+export interface INode extends Object {
+	id: {
+		name: string;
+	};
+	arguments: string[];
+	body: INode[];
+	elements: INode[];
+	expression: {
+		left: {
+			computed: boolean,
+			object: INode,
+			property: INode,
+			type: string,
+		},
+		operator: string,
+		right: INode,
+		type: string,
+	};
+	filter: (p: (p: INode) => boolean) => INode[] | any;
+	find: (objectExpression: object, filterExpression?: object) => INode;
+	forEach: (p: (p: INode) => void) => INode[] | any;
+	get: (property: string) => INode;
+	remove: () => void;
+	nodes: () => INode[];
+	pop: () => INode;
+	key: {
+		name: string;
+		value: INode | string | any;
+	};
+	node: INode;
+	name: string;
+	object: any;
+	parent: INode;
+	properties: INode[];
+	property: INode;
+	prune: Function;
+	replaceWith: (objectExpression: object) => INode;
+	size: () => number;
+	type: string;
+	value: INode | string | any;
+	toSource: (object: any) => string;
+}
+
+export interface IJSCodeshift extends Object {
+	(source?: INode | string): INode;
+	withParser: (parser: string) => IJSCodeshift;
+	identifier: (key: string) => INode;
+	literal: (key: string) => INode;
+	memberExpression: (node1: INode, node2: INode) => INode;
+	objectProperty: (key: INode, property: INode) => INode;
+	objectExpression: (properties: INode[]) => INode;
+	newExpression: (expression: INode, args: INode[]) => INode;
+	callExpression: (expression: INode, args: INode[]) => INode;
+	variableDeclarator: (key: INode, args: INode) => INode;
+	variableDeclaration: (key: string, args: INode[]) => INode;
+	arrayExpression: (args?: INode[]) => INode;
+	property: (type: string, key: INode, value: INode) => INode;
+	Property: object;
+	NewExpression: object;
+	VariableDeclarator: object;
+	Identifier: object;
+	Literal: object;
+	ArrayExpression: object;
+	MemberExpression: object;
+	FunctionExpression: object;
+	ObjectExpression: object;
+	BlockStatement: object;
+	Program: object;
+	filters: {
+		VariableDeclarator: {
+			requiresModule: Function,
+		},
+	};
+}
+
+export type valueType = string | number | boolean | object | any[] | any;
```

Comment:
nice, but remove any here and we can rather add the type to it later if it's throwing an error

Reply:
There are > 15 errors if we remove `any`. The reason is that compiler can't predict which one of the types it would be. We would have to explicitly check typeof everywhere in the code.

![image](https://user-images.githubusercontent.com/5961873/42185502-bf0d3228-7e66-11e8-9594-679a8ef9d11d.png)

Reply:
If you need to do that, then you need to do that. No shortcuts when that's the reason we have types. It could be a regexp